test/integration/backward_compatible/serial/CompactRestartTest.js

```diff
@@ -50,11 +50,14 @@ describe('CompactRestartTest', function() {
     before(async function () {
         TestUtil.markClientVersionAtLeast(this, '5.1.0');
         const comparisonValueForServerVersion520 = await TestUtil.compareServerVersionWithRC(RC, '5.2.0');
-        if ((await TestUtil.compareServerVersionWithRC(RC, '5.1.0')) < 0) {
-            this.skip();
-        }
-        // Compact serialization 5.2 server is not compatible with clients older than 5.2
-        if (comparisonValueForServerVersion520 >= 0 && !TestUtil.isClientVersionAtLeast('5.2.0')) {
+        const isClientVersionAtLeast520 = TestUtil.isClientVersionAtLeast('5.2.0');
+        // Compact serialization 5.2 and newer server is not compatible with clients older than 5.2
+        // Compact serialization 5.2 and newer clients are not compatible with servers older than 5.2
+        const isNotCompactCompatible =
+            (comparisonValueForServerVersion520 >= 0 && !isClientVersionAtLeast520) ||
+            (comparisonValueForServerVersion520 < 0 && isClientVersionAtLeast520);
+
+        if (isNotCompactCompatible) {
```

Comment:
Can you put this logic in TestUtil in a function? I think the name isCompactCompatible should be ok.